### ISSUE TITLE
minimum_should_match on target

### ISSUE THREAD

**Issue by borissmidt** [elastic/elasticsearch#22404]
**Describe the feature**:
ElasticSearch has a minimum_should_match which allows us to set the minimum amount of text that needs to be matched before it is counted as a match. 

It might be usefull to have a minimum_should_match for the targeting document. Where you can specify how much needs to match but depending on the target, other doucment its field. 

**example**

  ```
 document 1: {
     "productDescription": "this is an article about computer related things, for example hp computers screens"  
   }

   target document1: {
     "product": "hp computer "
   }

   target document2 : {
     "product": "hp laptop "
   }

   target document3 : {
     "product": "dell computer screen"
   }

    "query" : {
      "must": match{
        "product": "this is an article about computer related things, for example hp computers screens"  
      }
   }
```
Now if i execute this query i will get document 3 then 1 then 2. 

But i would like to be able to set the target match percentage. This would enable me to say that it has to match a certain percentage of the words of the target. 
Then a 100% match would only return document 1 and not 3 because it isn't a perfect match.

**Comment by jpountz** [comment]
I think it would be possible to do this efficiently, but we do not have a query that allows to do this at the moment.

I'm curious what your use-case actually is? Is it to build some kind of alert system when new documents are added that fully match a query?

**Comment by jpountz** [comment]
I opened an issue on the Lucene side too: https://issues.apache.org/jira/browse/LUCENE-7621.

**Comment by borissmidt** [comment]
A good example is document classification with matching.

I have a lot of unknown documents in my index(A). i want to find the most relevant document from an other index(B), Index B is categorized for example articles about computer programming is tagged with _computer programming_ but a more general article about computers is tagged with _computer_. I'm only allowed to find 1 match for each document in index A. Thus i have to do a match from A to B. 

I made a work around to this problem in my own code. By extracting my list of categories from the text of the document and then use it as a terms query on an non analysed field. This works perfectly at for my use case.

I cannot do the categorization at index time because the categories can change.

**Comment by jpountz** [comment]
Implemented in #27145
